common/src/sql/dbinit.sql

```diff
@@ -875,10 +875,12 @@ CREATE TABLE omicron.public.image (
     time_deleted TIMESTAMPTZ,
 
     project_id UUID NOT NULL,
+    is_global BOOL NOT NULL,
```

Comment:
Given any boolean column, there is a substantial chance it eventually needs to have more than two values. Therefore in the interest of extensibility it is usually better to make it a two-valued enum instead.

Reply:
My initial inclination is to disagree given the context and usage but I'll consider it further. 

Enums come at a cost. There's a bit of boilerplate (new db type, new data store type, new view type, conversions between the two previous types) as well as the need to be defensive in their usage to ensure adding a new variant doesn't introduce a breaking change. Not understanding how it would expand could mean that we make the wrong choice in writing the query and end up in a similar awkward situation where we still have to do a migration before we can make an update to how the enum is used.